tests/Unit/DelayRequestTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+use Saloon\Contracts\Response;
+use Saloon\Http\Faking\MockClient;
+use Saloon\Contracts\PendingRequest;
+use Saloon\Http\Faking\MockResponse;
+use GuzzleHttp\Promise\RejectedPromise;
+use GuzzleHttp\Promise\PromiseInterface;
+use Saloon\Exceptions\Request\RequestException;
+use Saloon\Tests\Fixtures\Requests\UserRequest;
+use Saloon\Tests\Fixtures\Exceptions\TestResponseException;
+
+test('async request delay works', function () {
+    $request = new UserRequest;
+    $request->delay()->set(1000);
+
+    expect($request->delay()->all())->toEqual(1000);
+
+    $start = microtime(true);
+    connector()->sendAsync($request)->wait();
+    expect(microtime(true) - $start)->toBeGreaterThan(1);
+
+    $mockClient = new MockClient([
+        MockResponse::make(['name' => 'Sam']),
+    ]);
+
+    $start = microtime(true);
+    connector()->sendAsync($request,$mockClient)->wait();
+    expect(microtime(true) - $start)->toBeGreaterThan(1);
+});
+
+test('request delay works', function () {
+    $request = new UserRequest;
+    $request->delay()->set(1000);
+
+    expect($request->delay()->all())->toEqual(1000);
+
+    $start = microtime(true);
+    connector()->send($request);
+    expect(microtime(true) - $start)->toBeGreaterThan(1);
+
+    $mockClient = new MockClient([
+        MockResponse::make(['name' => 'Sam']),
+    ]);
+
+    $start = microtime(true);
+    connector()->send($request,$mockClient);
+    expect(microtime(true) - $start)->toBeGreaterThan(1);
+});
+
```

Comment:
Please could we also write some additional tests based on my feedback above?

- Testing the delay works when on a connector
- Testing the delay works when both connector and request (should we merge or make the request's delay take priority?)
- Unit tests for the IntegerBodyRepository, making sure the methods all work as expected (I can do this if you want me to)

I also noticed these tests use mocking which means we're not testing the Guzzle implementation ,please can we also write a sync/async test that sends a real request with Guzzle?

Reply:
- I'll test delays on connectors aswell
- i'll add a test that checks both (request takes priority indeed).
- I'll add unit tests for intrepo

No i tested both the simulation sender & guzzlesender.